projects/vdk-plugins/vdk-impala/src/vdk/plugin/impala/templates/load/fact/insert/02-handle-quality-checks.py

```diff
@@ -0,0 +1,79 @@
+# Copyright 2021-2023 VMware, Inc.
+# SPDX-License-Identifier: Apache-2.0
+import os
+
+from vdk.api.job_input import IJobInput
+from vdk.plugin.impala.templates.utility import align_stg_table_with_target
+from vdk.plugin.impala.templates.utility import get_file_content
+from vdk.plugin.impala.templates.utility import get_staging_table_name
+
+SQL_FILES_FOLDER = (
+    os.path.dirname(os.path.abspath(__file__)) + "/02-requisite-sql-scripts"
+)
+
+"""
+This step is intened to handle quality checks if such provided
+and stop the data from being populated into the target table if the check has negative outcome.
+Otherwise the data will be directly processed according to the used template type
+"""
+
+
+def run(job_input: IJobInput):
+    job_arguments = job_input.get_arguments()
+
+    check = job_arguments.get("check")
+    partition_clause = job_arguments["_vdk_template_insert_partition_clause"]
+    source_schema = job_arguments.get("source_schema")
+    source_view = job_arguments.get("source_view")
+    target_schema = job_arguments.get("target_schema")
+    target_table = job_arguments.get("target_table")
+    insert_query = get_file_content(SQL_FILES_FOLDER, "02-insert-into-target.sql")
+
+    if check:
```

Comment:
You should notice al ot of similarties between this and https://github.com/vmware/versatile-data-kit/pull/2040/files 

I think it's good idea to consider a refactoring change (could be separate PR) where we remove some of the duplcaiton : 

For example we can introduce some helper class ChecksHelper where we can have method like 
```
def perform_quality_checks_and_staging(job_input: IJobInput, check, original_transform_query, [potentially other common args] ): 
``` 

where a lot of the logic about performing the check,c creating the staging tales and so on can be moved. 

What do you think? That would reduce the code in each of those classes by moving it to a common place. And make it easier to maintain and fix in the future

Reply:
I'll need some more time to come up with the best possible solution for this common functionality improvement as all of the templates have some specific implementation parts. The common between them is:
1) Staging table name is being calculated
2) Staging table is being aligned with the target one
3) All/part of the data is moved to the staging table
4) Checks are performed on the staging table/staging+target tables
5) Staging data is moved to target

Reply:
You do not have to address this comment in this PR. Let's merge it , and you can address in separate PR.